File: app/src/main/java/com/shafa/ali/kavir_msg/db/models/Post.java
package com.shafa.ali.kavir_msg.db.models;

import com.google.gson.annotations.SerializedName;

import io.realm.RealmObject;
import io.realm.annotations.Required;

public class Post extends RealmObject {
    @Required
    private String id;
    private String title;
    private String content;
    private String date;
    private String author;
    private String url;

    public Post() {
    }

    public Post(String id, String title, String content, String date, String author, String url) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.date = date;
        this.author = author;
        this.url = url;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }
}

File: app/src/main/java/com/kavirelectronic/ali/kavir_info/db/models/Post.kt
package com.kavirelectronic.ali.kavir_info.db.models

import io.realm.RealmObject
import io.realm.annotations.Required

open class Post : RealmObject {
    @Required
    var id: String? = null
    var title: String? = null
    var content: String? = null
    var date: String? = null
    var author: String? = null
    var url: String? = null

    constructor() {}
    constructor(id: String?, title: String?, content: String?, date: String?, author: String?, url: String?) {
        this.id = id
        this.title = title
        this.content = content
        this.date = date
        this.author = author
        this.url = url
    }

}